File: 9041 - Software Instruction/lab06/whale.py
#!/usr/bin/python

import sys,re

whale=sys.argv[1]
pod=0
number=0
for string in sys.stdin.readlines():
	if re.search(whale,string):
		strList = re.split(r"\s+",string)
		pod = pod + 1
		number = number + int(strList[0])

print whale + " observations: " + str(pod) + " pods, " + str(number) +" individuals"

File: 9041 - Software Instruction/lab06/whale.pl
#!/usr/bin/perl -w

@input=<STDIN>;
$whale=$ARGV[0];

$pod=0;
$number=0;
foreach my $string (@input)
{
	if ($string =~ /$whale/) {
		my($count, $name) = split / /, $string;
		$pod++;
		$number = $number + $count;
	}
}

print "$whale observations: $pod pods, $number individuals\n";
